src/service/url-replacements-impl.js

```diff
@@ -163,6 +164,14 @@ export class UrlReplacements {
       return removeFragment(info.sourceUrl);
     }));
 
+    this.setAsync_('SOURCE_URL', () => {
+      return getTrackImpressionPromise().then(
+        this.getDocInfoValue_.bind(this, info => {
```

Comment:
I think there is a problem here:

Doc info eagerly reads `location.href` and never reads it again after.